src/crimson/osd/pg_shard_manager.h

```diff
@@ -432,6 +434,48 @@ class PGShardManager {
     return std::make_pair(id, std::move(fut));
   }
 
+  template <typename T, typename... Args>
+  auto start_pg_operation_active(Args&&... args) {
```

Comment:
How can we know that users of this do not need to go through the usual stages/checks?
Should we assert that the target pg is at `opref.get_epoch()` for example?

Reply:
The messages in question are only sent between replicas and primaries after they are already active.